.github/workflows/tests.yml

```diff
@@ -105,6 +149,25 @@ jobs:
         run: git fetch origin ${GITHUB_REF}:ci-branch
       - name: "Check out branch"
         run: git checkout --force ci-branch
+      # --- Cache based on Conan version and ExternalProjects cmake source
+      - name: "Get Conan version"
+        id: get-conan-version
+        run: |
+          echo "::set-output name=conan-version::$(conan --version | tr -d '[:alpha:] ')"
+        shell: bash
+      - name: "Hash ExternalProjects"
+        id: hash-external
+        run: |
+          echo "::set-output name=hash-external::$(sha256sum cmake/ExternalProjects.cmake | cut '-d ' -f 1)"
+        shell: bash
+      - uses: actions/cache@v2
+        with:
+          path: '~/.conan'
+          key: ${{ runner.os }}-${{ steps.get-conan-version.outputs.conan-version }}-${{ steps.hash-external.outputs.hash-external }}-release
```

Comment:
`-release` Should be at the beginning of the key, as the restore-keys matches any keys starting with the given strings, so it would restore the debug key instead of forcing a fresh release key.

Reply:
To clarify, we want a completely different key space for the `release` versions, so we'd have:

```
key: release-${{ runner.os }}-${{ steps.get-conan-version.outputs.conan-version }}-${{ steps.hash-external.outputs.hash-external }}
restore-keys: |
  release-${{ runner.os }}-${{ steps.get-conan-version.outputs.conan-version }}-
  release-${{ runner.os }}-
```